docs/rst/tutorials/core/deployment/keys/topic_keys.rst

```diff
@@ -0,0 +1,234 @@
+.. include:: ../../../../exports/alias.include
+
+.. _topic_keys_tutorial:
+
+Topic Keys Tutorial
+===================
+
+.. contents::
+    :depth: 2
+    :local:
+    :backlinks: none
+
+Background
+----------
+
+In Vulcanexus, a *Topic* is a communication channel used for publishing and subscribing to messages.
+The value of data associated with a topic changes over time and each of these values are known as *data samples*.
+:ref:`topic_keys` refer to topics where each data sample is uniquely identified by a set of key fields (known as *data instance*).
+
+Unlike standard topics, where each data sample is distinct, keyed topics allow the user to manage and
+access data samples more efficiently by associating them with specific keys.
+Please, refer to the documented section on :ref:`topic_keys` for a more detailed explanation.
+
+Creating custom IDL messages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In ROS 2, users can define its own :ref:`custom messages<CustomInterfaces>`. In addition,
+a user can also create custom messages using the `IDL (Interface Definition Language) <https://www.omg.org/spec/IDL/4.2/About-IDL>`_
+format following a standardized way of defining the message structure.
+Next example depicts how to define a custom message using the IDL format in ROS 2:
+
+.. code-block:: bash
+
+    # ExampleMsg.idl
+    module package_name {
+      module msg {
+        struct ExampleMsg {
+          long field;
+          short another_field;
+          boolean yet_another_field;
+        };
+      };
+    };
+
+As it can be seen, it is required to name the upper module with the same name as the package name containing it and
+with a nested module called *msg*. Within the *msg* module, user can define the message structure as desired using the IDL syntax
+and name it with the same name as the file.
+
+One of the advantages of defining messages in Interface Definition Language (IDL) is the ability to use annotations.
+Annotations are metadata that provide additional information about IDL constructs such as modules,
+interfaces, operations, attributes, and data types and are used to convey details that are not part of the core
+IDL syntax but are relevant for code generation, documentation, or other purposes.
+Annotations in IDL typically follow the ``@`` symbol and can be applied to various IDL constructs.
+
+Creating Keyed Messages
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Keyed topics typically consist of one or more key fields within the data sample structure.
```

Comment:
They do not consists of key fields per-se. Keyed topics exist when one or more fields in the data structure are annotated as keys